### PR TITLE
join ffbs-mesh-vpn-parker with ffbs-parker-nextnode

### PR DESCRIPTION
**Pull request by maurerle**
This adds requirements to ffbs-mesh-vpn-parker to include the other parker-related packages.

It should be possible to build it with the state of https://github.com/ffac/gluon/tree/v2023.2.x-parker which did revert the mesh-vpn related commits from the parker rebase.

A site-configuration then only needs to require `ffbs-mesh-vpn-parker` to work when build from a parker gluon branch.

This change is backward compatible by providing the former name in the Makefile as well

Note: not tested yet

### CODE REVIEW

**Comment by SmithChart** [comment]
Thx for the input. For some reason I did not see the mail notification for this PR. I will try to integrate this changes into my PRs on community-packages.

File: ffbs-mesh-vpn-parker/luasrc/usr/share/lua/nodeconfig.lua
Comment (SmithChart) [inline ffbs-mesh-vpn-parker/luasrc/usr/share/lua/nodeconfig.lua:1]: I am not a fan of having these files get minimized. This makes understanding and debugging in the target unnecessary complex. I would vote to leave `noderoute.lua` and `nodeconfig.lua` outside of `luasrc`.

File: ffbs-mesh-vpn-parker/luasrc/usr/share/lua/noderoute.lua
Comment (SmithChart) [inline ffbs-mesh-vpn-parker/luasrc/usr/share/lua/noderoute.lua:1]: I am not a fan of having these files get minimized. This makes understanding and debugging in the target unnecessary complex. I would vote to leave `noderoute.lua` and `nodeconfig.lua` outside of `luasrc`.

File: ffbs-parker-nodeconfig/Makefile [hunk 1]
```diff
@@ -21,19 +21,5 @@ define Package/ffbs-parker-nodeconfig/description
...
-
-define Build/Compile
-endef
-
-define Package/ffbs-parker-nodeconfig/install
-	$(CP) ./files/* $(1)/
-endef
-
-$(eval $(call BuildPackage,ffbs-parker-nodeconfig))
+$(eval $(call BuildPackageGluon,ffbs-parker-nodeconfig))
```
Comment (SmithChart) [inline ffbs-parker-nodeconfig/Makefile]: Is `BuildPackageGluon` preferable over `BuildPackage`? What's the difference?

File: ffbs-parker-nodeconfig/Makefile [hunk 2]
```diff
@@ -8,11 +8,11 @@ PKG_LICENSE:=MIT
...
-include $(INCLUDE_DIR)/package.mk
+include $(TOPDIR)/../package/gluon.mk
 
 define Package/ffbs-parker-nodeconfig
   TITLE:=Automatic node configuration for parker networks
-  DEPENDS:=+kmod-wireguard +wireguard-tools +luaposix +usign +gluon-radv-filterd +gluon-ebtables-filter-ra-dhcp
+  DEPENDS:=+gluon-core +libgluonutil +gluon-mesh-vpn-core +kmod-wireguard +wireguard-tools +luaposix +usign +gluon-radv-filterd +gluon-ebtables-filter-ra-dhcp
 endef
 
 define Package/ffbs-parker-nodeconfig/description
```
Comment (SmithChart) [inline ffbs-parker-nodeconfig/Makefile]: Unrelated change to VPN provider? Please split commit or comment.

File: ffbs-parker-nodeconfig/Makefile (hunk 1 above)
Comment (SmithChart) [inline ffbs-parker-nodeconfig/Makefile]: Unrelated change to VPN provider? Please split commit or comment.

File: ffbs-mesh-vpn-parker/README.md
Comment (SmithChart) [inline ffbs-mesh-vpn-parker/README.md:1]: Readme contains the package name. Should be renamed there, too.

**Comment by SmithChart** [comment]
Can you rebase onto my recent topic branch?

File: ffbs-mesh-vpn-parker/luasrc/usr/share/lua/nodeconfig.lua
Comment (maurerle) [reply ffbs-mesh-vpn-parker/luasrc/usr/share/lua/nodeconfig.lua:1]: I can understand your concerns. Though I don't think that it is useful for all compiled targets.
Eventually there is a setting to disable this in openwrt for debug builds?

I for myself have always found what I needed from the minimized outputs and this is better in >99% of devices.

File: ffbs-parker-nodeconfig/Makefile (hunk 1 above)
Comment (maurerle) [reply ffbs-parker-nodeconfig/Makefile]: BuildPackageGluon does also include the check_site checks for gluon, sets tags and handles the respondd.c part directly:
https://github.com/freifunk-gluon/gluon/blob/main/package/gluon.mk#L91

Other than that, it is not required, but I would use it for consistency with the other packages as this package relies on gluon anyway

File: ffbs-mesh-vpn-parker/luasrc/usr/share/lua/nodeconfig.lua
Comment (SmithChart) [reply ffbs-mesh-vpn-parker/luasrc/usr/share/lua/nodeconfig.lua:1]: As discusses with @grische off-channel: I will drop this point. It seems other communities rely on saving these last bytes. So I don't want to be in their way.

File: ffbs-mesh-vpn-parker/luasrc/usr/share/lua/noderoute.lua
Comment (SmithChart) [reply ffbs-mesh-vpn-parker/luasrc/usr/share/lua/noderoute.lua:1]: As discusses with @grische off-channel: I will drop this point. It seems other communities rely on saving these last bytes. So I don't want to be in their way.

File: ffbs-parker-nodeconfig/Makefile (hunk 2 above)
Comment (maurerle) [reply ffbs-parker-nodeconfig/Makefile]: So this change is part of `add parker provider to ffbs-parker-nodeconfig`.
This ties it a little further towards gluon and also requires to have a gluon vpn available.
Therefore I think it is useful to have this change together.

Without the change to gluon.mk, the `check_site.lua` would not be executed.
Without `gluon-mesh-vpn-core` the additional `gluon/mesh-vpn/provider` file does not make much sense.
What do you think?

File: ffbs-parker-nodeconfig/Makefile (hunk 1 above)
Comment (maurerle) [reply ffbs-parker-nodeconfig/Makefile]: This makes sure that the check_site.lua is actually executed on build.
See also: https://gluon.readthedocs.io/en/latest/dev/packages.html#provided-macros

Comment (SmithChart) [reply ffbs-parker-nodeconfig/Makefile]: With your explanation it makes sense. Thx.

File: ffbs-parker-nodeconfig/Makefile (hunk 2 above)
Comment (SmithChart) [reply ffbs-parker-nodeconfig/Makefile]: I've now understand why this is related. Thx.

**Review by SmithChart — APPROVED** [review]
